barman/fs.py

```diff
@@ -48,25 +52,27 @@ def get_last_output(self):
         """
         return self.internal_cmd.out, self.internal_cmd.err
 
-    def create_dir_if_not_exists(self, dir_path):
+    def create_dir_if_not_exists(self, dir_path, mode=None):
         """
         This method recursively creates a directory if not exists
 
         If the path exists and is not a directory raise an exception.
 
         :param str dir_path: full path for the directory
+        :param mode str|None: Specify the mode to use for creation. Not used
+            if the directory already exists.
+        :returns bool: False if the directory already exists True if the
+            directory is created.
         """
         _logger.debug("Create directory %s if it does not exists" % dir_path)
-        exists = self.exists(dir_path)
```

Comment:
Nice - I didn't spot the duplication here with the `check_directory_exists` method.